packages/webdriverio/src/session/context.ts

```diff
@@ -177,4 +196,97 @@ export class ContextManager extends SessionManager {
     get mobileContext() {
         return this.#mobileContext
     }
+
+    /**
+     * Get the flat context tree for the current session
+     * @returns a flat list of all contexts in the current session
+     */
+    async getFlatContextTree (): Promise<Record<string, FlatContextTree>> {
+        const tree = await this.#browser.browsingContextGetTree({})
+
+        const mapContext = (context: local.BrowsingContextInfo): string[] => [
+            context.context,
+            ...(context.children || []).map(mapContext).flat(Infinity) as string[]
+        ]
+
+        /**
+         * transform context tree into a flat list of context objects with references
+         * to children
+         */
+        const allContexts: Record<string, FlatContextTree> = tree.contexts.map(mapContext).flat(Infinity)
+            .reduce((acc, ctx: string) => {
+                const context = this.findContext(ctx, tree.contexts, 'byContextId')
+                acc[ctx] = context as unknown as FlatContextTree
+                return acc
+            }, {} as Record<string, FlatContextTree>)
+        return allContexts
+    }
+
+    /**
+     * Find the parent context of a given context id
+     * @param contextId the context id you want to find the parent of
+     * @param contexts  the list of contexts to search through returned from `browsingContextGetTree`
+     * @returns         the parent context of the context with the given id
+     */
+    findParentContext (contextId: string, contexts: local.BrowsingContextInfoList): local.BrowsingContextInfo | undefined {
+        for (const context of contexts) {
+            if (context.children?.find((child) => child.context === contextId)) {
```

Comment:
I believe that using .some is a tiny bit cheaper to execute as we only need a Boolean to be returned instead of the actual item

Reply:
good catch!